presto-main/src/main/java/com/facebook/presto/server/CoordinatorModule.java

```diff
@@ -361,6 +362,9 @@ protected void setup(Binder binder)
         executionPolicyBinder.addBinding("all-at-once").to(AllAtOnceExecutionPolicy.class);
         executionPolicyBinder.addBinding("phased").to(PhasedExecutionPolicy.class);
 
+        configBinder(binder).bindConfig(NodeResourceStatusConfig.class);
+        binder.bind(NodeResourceStatusProvider.class).to(NodeResourceStatus.class).in(Scopes.SINGLETON);
```

Comment:
this injection should be in ServerMainModule and not here as it needs to be available to both Coordinator and Workers.

Reply:
missed the other injection for WorkerModule. resolved.